### PR TITLE
[fix][storage] Fix port conflict in bookkeeper.conf

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
`prometheusStatsHttpPort=8000` in the `bookkeeper.conf`
and` metricsProvider.httpPort=8000` in the `zookeeper.conf`, there will be a port conflict when deploying a cluster.

### Modification
Change the prometheusStatsHttpPort=8001

### Reproduce
1. modify the configuration in `broker.conf`

> metadataStoreUrl=127.0.0.1:2181
clusterName=cluster-a
managedLedgerDefaultEnsembleSize=1
managedLedgerDefaultWriteQuorum=1
managedLedgerDefaultAckQuorum=1

2. start zookeeper
```shell
bin/pulsar zookeeper.
```
3. init the cluster metadata
```shell
 bin/pulsar initialize-cluster-metadata \
          --cluster cluster-a \
          --zookeeper 127.0.0.1:2181 \
          --configuration-store 127.0.0.1:2181 \
          --web-service-url http://127.0.0.1:8080 \
          --broker-service-url pulsar://127.0.0.1:6650 \
```
4. start broker
```shell
bin/pulsar broker
```
5. start bookie
```shell
bin/pulsar bookie
``` 

## Screenshot
image

### Documentation

- [x] `doc-not-needed`

### CODE REVIEW

**Review by nodece — APPROVED** [review]
LGTM

**Comment by congbobo184** [comment]
it's better to change `httpServerPort`,  `httpServerEnabled` default is false.

**Comment by tisonkun** [comment]
@Anonymitaet @hangc0276 now we encounter the first PR title issue. `conf` is not in our scope candidates, if we don't sort this patch in to `conf`, shall we sort it into...`storage` (since it's config of bookkeeper)?

**Comment by Anonymitaet** [comment]
> @Anonymitaet @hangc0276 now we encounter the first PR title issue. `conf` is not in our scope candidates, if we don't sort this patch in to `conf`, shall we sort it into...`storage` (since it's config of bookkeeper)?
> 
> See the failure report at: https://github.com/apache/pulsar/actions/runs/3216489256/jobs/5258426561

`storage` makes sense to me because `conf` involves various areas, eg, `io conf`, `fn conf`, `client conf`, `cli conf`, ....
Each of them should belong to `io`, `fn`, `client`, `cli` rather than `conf`.

Details see [[Guide] Pulsar Pull Request Naming Convention](https://docs.google.com/document/d/1d8Pw6ZbWk-_pCKdOmdvx9rnhPiyuxwq60_TrD68d7BA/edit#bookmark=id.y8943h392zno) 
image

cc @D-2-Ed @DaveDuggins @momo-jun

File: conf/bookkeeper.conf [hunk 1]
```diff
@@ -694,7 +694,7 @@ httpServerEnabled=false
 
 # The http server port to listen on. Default value is 8080.
 # Use `8000` as the port to keep it consistent with prometheus stats provider
-httpServerPort=8000
+httpServerPort=8001
```
Comment (codelipenghui) [inline conf/bookkeeper.conf:697]: @liangyepianzhou If the bookie side enabled the http server, prometheus should also use the http server port, not a separate port. Can you reproduce the issue?

Comment (liangyepianzhou) [reply conf/bookkeeper.conf:697]: Thanks for your feedback!
When deploying the cluster before, there were often conflicts due to the port of prometheusStatsHttpPort=8000.
Then I found that there are two configurations whose port is 8000 in the bookkeeper.
If not for this reason I will redeploy the cluster and find out the real reason for the port conflict later.

Comment (codelipenghui) [reply conf/bookkeeper.conf:697]: If you are deploying the bookkeeper and zookeeper on the same node, it should conflict with the port used by Zookeeper

**Review by hangc0276 — CHANGES_REQUESTED** [review]
Does it happen in our BookKeeper cluster env? In BookKeeper, if the user enabled httpServer, the httpServerPort will override prometheusHttpPort and export Prometheus metrics through httpServerPort.

**Comment by liangyepianzhou** [comment]
@hangc0276 Thanks for your feedback!
 I'm rebasing to the latest version. 
I will redeploy the cluster after installation to see whether the problem can be reproduced.

**Comment by liangyepianzhou** [comment]
@hangc0276 
I deployed the Pulsar cluster with the latest master branch and got this exception.
Configurations in broker.conf:

> metadataStoreUrl=127.0.0.1:2181
> clusterName=cluster-a 
> managedLedgerDefaultEnsembleSize=1
> managedLedgerDefaultWriteQuorum=1  
> managedLedgerDefaultAckQuorum=1

And the exception is 

> 2022-10-14T13:52:34,154+0800 [main] ERROR org.apache.bookkeeper.common.component.AbstractLifecycleComponent - Failed to start Component: stats-provider
java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:8000
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:119) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.service.StatsProviderService.doStart(StatsProviderService.java:51) ~[bookkeeper-server-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:83) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.LifecycleComponentStack.lambda$start$4(LifecycleComponentStack.java:144) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[guava-31.0.1-jre.jar:?]
        at org.apache.bookkeeper.common.component.LifecycleComponentStack.start(LifecycleComponentStack.java:144) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.ComponentStarter.startComponent(ComponentStarter.java:85) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.Main.doMain(Main.java:280) ~[bookkeeper-server-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.Main.main(Main.java:255) ~[bookkeeper-server-4.15.1.jar:4.15.1]
Caused by: java.io.IOException: Failed to bind to /0.0.0.0:8000
        at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.Server.doStart(Server.java:401) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:116) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        ... 8 more
Caused by: java.net.BindException: Address already in use
        at sun.nio.ch.Net.bind0(Native Method) ~[?:?]
        at sun.nio.ch.Net.bind(Net.java:555) ~[?:?]
        at sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337) ~[?:?]
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294) ~[?:?]
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89) ~[?:?]
        at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:344) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.Server.doStart(Server.java:401) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:116) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        ... 8 more
2022-10-14T13:52:34,301+0800 [main] ERROR org.apache.bookkeeper.common.component.AbstractLifecycleComponent - Calling uncaughtExceptionHandler
2022-10-14T13:52:34,302+0800 [main] ERROR org.apache.bookkeeper.common.component.ComponentStarter - Triggered exceptionHandler of Component: bookie-server because of Exception in Thread: Thread[main,5,main]
java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:8000
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:119) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.service.StatsProviderService.doStart(StatsProviderService.java:51) ~[bookkeeper-server-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:83) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.LifecycleComponentStack.lambda$start$4(LifecycleComponentStack.java:144) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[guava-31.0.1-jre.jar:?]
        at org.apache.bookkeeper.common.component.LifecycleComponentStack.start(LifecycleComponentStack.java:144) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.common.component.ComponentStarter.startComponent(ComponentStarter.java:85) ~[bookkeeper-common-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.Main.doMain(Main.java:280) ~[bookkeeper-server-4.15.1.jar:4.15.1]
        at org.apache.bookkeeper.server.Main.main(Main.java:255) ~[bookkeeper-server-4.15.1.jar:4.15.1]
Caused by: java.io.IOException: Failed to bind to /0.0.0.0:8000
        at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.Server.doStart(Server.java:401) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:116) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        ... 8 more
Caused by: java.net.BindException: Address already in use
        at sun.nio.ch.Net.bind0(Native Method) ~[?:?]
        at sun.nio.ch.Net.bind(Net.java:555) ~[?:?]
        at sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337) ~[?:?]
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294) ~[?:?]
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89) ~[?:?]
        at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:344) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.server.Server.doStart(Server.java:401) ~[jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73) ~[jetty-util-9.4.48.v20220622.jar:9.4.48.v20220622]
        at org.apache.bookkeeper.stats.prometheus.PrometheusMetricsProvider.start(PrometheusMetricsProvider.java:116) ~[prometheus-metrics-provider-4.15.1.jar:4.15.1]
        ... 8 more

**Comment by hangc0276** [comment]
> @hangc0276 Thanks for your feedback! I'm rebasing to the latest version. I will redeploy the cluster after installation to see whether the problem can be reproduced.

Zookeeper Prometheus metrics use `8000`, does the BK Prometheus metric conflict with ZK?

**Comment by liangyepianzhou** [comment]
> Zookeeper Prometheus metrics use 8000, does the BK Prometheus metric conflict with ZK?
Yes, It is the BK Prometheus metric port that conflict with the ZK Prometheus metric port.
Should we modify one of them or make one of them with no default value?

**Comment by hangc0276** [comment]
> > Zookeeper Prometheus metrics use 8000, does the BK Prometheus metric conflict with ZK?
> 
> Yes, It is the BK Prometheus metric port that conflict with the ZK Prometheus metric port. Should we modify one of them or make one of them with no default value?

I prefer not to modify anything. Most of the cases, we won't deploy ZK and bookie in the same node.  If users want to do that, he just needs to change one of the port numbers

**Comment by liangyepianzhou** [comment]
> I prefer not to modify anything. Most of the cases, we won't deploy ZK and bookie in the same node. If users want to do that, he just needs to change one of the port numbers

OK, Thanks for your feedback!
I will close this PR.

**Pull request closed without merging**